src/lib/modules/blockchain_connector/index.js

```diff
@@ -699,6 +702,26 @@ class BlockchainConnector {
       callback();
     });
   }
+
+  _getEvents() {
+    const data = this._readEvents();
+    return Object.values(data).reverse();
+  }
+
+  _saveEvent(event) {
+    const data = this._readEvents();
+    data[new Date().getTime()] = event;
+    fs.writeJSONSync(this.logFile, data);
```

Comment:
If this will result in a lot of sync fs access, should we use async (`writeJson`) instead of sync? We'd need to save events to a queue, drain/wait, then loop back. I don't want things to be over-complicated but I'd hate to accidentally started bogging down embark.

Reply:
I feel it would indeed be over-complicated.
Also those actions are done when embark is doing nothing but waiting.

The code-generator perform a lot more file system action without using such technique and performance is more than ok